llama-index-integrations/llms/llama-index-llms-openai/llama_index/llms/openai/base.py

```diff
@@ -946,6 +946,10 @@ def get_tool_calls_from_response(
             except ValueError:
                 argument_dict = {}
 
+            # Treat parsable non-dict JSON as empty
+            if not isinstance(argument_dict, dict):
+                argument_dict = {}
+
             tool_selections.append(
                 ToolSelection(
```

Comment:
Couldn't we just update the `ToolSelection` object to default to an empty dict? Then it generalizes to every LLM

Reply:
OK, I'll make that change.